kotlin-native/backend.native/tests/interop/objc/kt65260.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2010-2024 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+import platform.Foundation.NSArray
+import platform.Foundation.NSString
+import platform.darwin.NSObject
+
+fun main() {
+    // Test: cast to NSObject Metaclass
+    Any() as NSObject.Companion
+    // Test: cast to subclass of NSObject Metaclass
+    Any() as NSString.Companion
+    Any() as NSArray.Companion
```

Comment:
All these casts should fail by definition. Do they?

For example, `NSObject.Companion` is a companion object class, meaning that the only instance of this class is the companion object. In other words, `NSObject as NSObject.Companion` should succeed, but `Any() as NSObject.Companion` should fail (i.e. throw an exception).

Reply:
Yes, all these tests will throw `kotlin.TypeCastException` in runtime.
```
Any() as NSObject.Companion

Uncaught Kotlin exception: kotlin.TypeCastException: class kotlin.Any cannot be cast to class platform.darwin.NSObject.Companion
```

`NSObject as NSObject.Companion` compile works, but it‘s unnecessary, also IDEA prompts `No cast needed` warning.
```kotlin
// Two expressions are equivalent
val nsObject = NSObject
val nsObjectComponion = NSObject.Companion
```

Reply:
> `NSObject as NSObject.Companion` compile works, but it‘s unnecessary, also IDEA prompts `No cast needed` warning.

Checking this is still necessary, because the compiler can't always rely on compile-time types. For example,
```kotlin
val nsObjectClass: Any = NSObject
nsObjectClass as NSObject.Companion
```
In this case, the compiler will generate the actual cast.